File: exercise/14/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string result = "";
        if (strs.size() == 0) {
            return result;
        }
        int pos = 0;
        while (true) {
            char now;
            if (pos < strs[0].length()) {
                now = strs[0][pos];
            } else {
                return result;
            }
            for (int i = 1; i < strs.size(); i++) {
                if (pos < strs[i].length() && now == strs[i][pos]) {
                    continue;
                }
                return result;
            }
            result += now;
            pos++;
        }
        return result;
    }
};

File: exercise/14/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        result = ""
        for i in range(0, len(strs[0])):
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                    return result
            result += strs[0][i]
        return result
